src/cli/commands/lint.ts

```diff
@@ -70,20 +78,15 @@ linting ./openapi.yaml
 }
 
 async function lint(name: string, flags: any, command: Lint) {
-  command.log(`linting ${name}`);
-  let obj: IParserResult;
-  try {
-    obj = await readInputArguments(name, flags.encoding);
-  } catch (ex) {
-    throw new Error(`Could not parse ${name}: ${ex.message}`);
-  }
+  command.log(`Linting ${name}`);
+  const spec: IParserResult = await readParsable(name, flags.encoding);
 
   const spectral = new Spectral();
-  if (obj.data.swagger && obj.data.swagger === '2.0') {
+  if (spec.data.swagger && spec.data.swagger === '2.0') {
     command.log('OpenAPI 2.0 (Swagger) detected');
     spectral.addFunctions(oas2Functions());
     spectral.addRules(oas2Rules());
-  } else if (obj.data.openapi && typeof obj.data.openapi === 'string' && obj.data.openapi.startsWith('3.')) {
+  } else if (spec.data.openapi && typeof spec.data.openapi === 'string' && spec.data.openapi.startsWith('3.')) {
```

Comment:
You can use `parseInt` instead of string comparison, it's going to detect anything `3.x.y`

```bash
> parseInt('3.1.2')
3
```